### PR TITLE
time synch now checks message timestamp

### PR DESCRIPTION
**Pull request by GunnarRolander**
Time synch tests now check StatusRequest timestamps as mentioned in #70

### CODE REVIEW

**Comment by emiltin** [comment]
reset_date() in command_helpers.rb must use UTC time. RSMP uses UTC for all timestamps.

There's might also be an issue with using in inside an ensure block, since it involves expect check, and might therefore raise exceptions.

**Comment by sweco-seguro** [comment]
What is up with this error?
´´´RSMP::NotReady:
       RSMP::NotReady´´´

I get it quite often when running against the rsmp gem TLC.

File: spec/support/command_helpers.rb [hunk 1]
```diff
@@ -271,18 +271,20 @@ def set_security_code level
...
+      month: date.month,
+      day: date.day,
+      hour: date.hour,
+      minute: date.min,
+      second: date.sec
     }
-    send_command_and_confirm @task, command_list, "intention to set date"
+
+    message, result = send_command_and_confirm @task, command_list, "intention to set date"
+    return message, result
```
Comment (emiltin) [inline spec/support/command_helpers.rb:287]: the two lines can be reduced to just:
send_command_and_confirm @task, command_list, "intention to set date"

File: spec/support/command_helpers.rb [hunk 2]
```diff
@@ -291,12 +293,12 @@ def reset_date
     @site.log "Reset date", level: :test
     command_list = build_command_list :M0104, :setDate, {
       securityCode: SECRETS['security_codes'][1],
-      year: Time.now.year,
-      month: Time.now.month,
-      day: Time.now.day,
-      hour: Time.now.hour,
-      minute: Time.now.min,
-      second: Time.now.sec
+      year: Time.now.utc.year,
```
Comment (emiltin) [inline spec/support/command_helpers.rb:296]: better to use Time.now.utc once and store in a local, then access year, month, etc in that.

**Comment by emiltin** [comment]
>      RSMP::NotReady:

It means the site did not complete the connction sequence within the expected time.
You can try to increase the ready_timeout in config/rsmp_gem.yaml. It's currently set to 1s which works on my machine, but maybe you need a higher value. Try setting it to 10s just to see if that is the cause of the problem.

**Comment by emiltin** [comment]
I was to quick about NotReady. In fact it's raised by methods like request_status() if the proxy is not ready, ie. there is no connection yet or the handshake is not complete yet. Not sure why you get it, as each time you use e.g. TestSite.connection, it should wait for ready state in this bit:
```ruby
    @remote_site.wait_for_state :ready, RSMP_CONFIG['ready_timeout']
```

**Comment by emiltin** [comment]
When there are changes in master, I like to rebase on master, rather then merge. ie:
```
git rebase master
```
That will replay your commits on top of the latest master. You commits will be kept together and there will be no merge messages in the commit history.

**Comment by emiltin** [comment]
Hi Hunnar, I've pushed some changed.
There's a few tricky issues.
a) request_status() returns [request, result]. Requst is the message send. Result is not a message, but a hash with the collected status values. This is because the response can be split into multiple messages. This means that if we want to look at the RSMP timestamp, we need an actual response messge, and the collect() methods don't all return that currently. No big deal, since we collect it outself 'Check status response timestamp', but I will consider making it easier to get to the messages from the collect methods.

b) response.timestamp is an internal timestamp which is set when the mesages is received, using the system clock on the local machine. The RSMP timestamp is in response.attributes['sTs'] for status responses, but the name varies (enfortunetely) between messages.

c) the date cannot be reset using an ensure block in the set_date() helpers, otherwise the date will be reset before we can do our checks. I added a small helper with_date_sets with handles the resetting.

d) i've uddated the rsmp gem. it now uses the time set with m0104 for the various mesage timestamp. (i think watchdogs might be missing). all the tests currently pass

**Comment by emiltin** [comment]
I added that last to two tests. the gem will now return the actual rsmp messages when collecting, in addition to the attribute values.

There's some aspects of how excetions are handles which is a bit akward when returning multiple values, so i might do another refactoring of the collect methods at some point.

But for now, it's a bit easier to check the content of actual rsmp messages collected. It also means we don't need to do any manual collection (and task handling) in these clock specs.

**Pull request merged**